File: examples/local-private-notes/app/policy-demo.mjs
import assert from "node:assert/strict";
import { evaluateAxiomPolicy } from "../generated/policy-evaluator.mjs";

const capability = "summarize_private_note";

const scenarios = [
  {
    name: "local summary",
    expected: "allow",
    facts: {
      owner_authenticated: true,
      note_selected: true,
      destination_local: true,
    },
  },
  {
    name: "external destination",
    expected: "require_approval",
    facts: {
      destination_external: true,
    },
  },
  {
    name: "raw note request",
    expected: "deny",
    facts: {
      owner_authenticated: true,
      note_selected: true,
      destination_local: true,
      requests_raw_note: true,
    },
  },
];

const results = scenarios.map((scenario) => {
  const result = evaluateAxiomPolicy(capability, scenario.facts);
  assert.equal(result.decision, scenario.expected);
  return {
    scenario: scenario.name,
    decision: result.decision,
    reasons: result.reasons,
  };
});

console.log(JSON.stringify(results, null, 2));

File: examples/local-private-notes-python/app/policy_demo.py
import json
import sys
from pathlib import Path


generated_dir = Path(__file__).resolve().parents[1] / "generated-python"
sys.path.insert(0, str(generated_dir))

from policy_evaluator import evaluate_axiom_policy  # noqa: E402


CAPABILITY = "summarize_private_document"

SCENARIOS = [
    {
        "name": "local summary",
        "expected": "allow",
        "facts": {
            "owner_authenticated": True,
            "document_selected": True,
            "destination_local": True,
        },
    },
    {
        "name": "external destination",
        "expected": "require_approval",
        "facts": {
            "destination_external": True,
        },
    },
    {
        "name": "raw document request",
        "expected": "deny",
        "facts": {
            "owner_authenticated": True,
            "document_selected": True,
            "destination_local": True,
            "requests_raw_document": True,
        },
    },
]


def main() -> None:
    results = []
    for scenario in SCENARIOS:
        result = evaluate_axiom_policy(CAPABILITY, scenario["facts"])
        assert result["decision"] == scenario["expected"], result
        results.append(
            {
                "scenario": scenario["name"],
                "decision": result["decision"],
                "reasons": result["reasons"],
            }
        )

    print(json.dumps(results, indent=2))


if __name__ == "__main__":
    main()
